iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java

```diff
@@ -540,6 +540,43 @@ public void testCTASFromHiveTable() {
     Assert.assertArrayEquals(new Object[]{2L, "Linda", "Finance"}, objects.get(1));
   }
 
+  @Test
+  public void testCTASPartitionedFromHiveTable() throws TException, InterruptedException {
+    Assume.assumeTrue("CTAS target table is supported fully only for HiveCatalog tables." +
+        "For other catalog types, the HiveIcebergSerDe will create the target Iceberg table in the correct catalog " +
+        "using the Catalogs.createTable function, but will not register the table in HMS since those catalogs do not " +
+        "use HiveTableOperations. This means that even though the CTAS query succeeds, the user would not be able to " +
+        "query this new table from Hive, since HMS does not know about it.",
+        testTableType == TestTables.TestTableType.HIVE_CATALOG);
+
+    shell.executeStatement("CREATE TABLE source (id bigint, name string) PARTITIONED BY (dept string) STORED AS ORC");
+    shell.executeStatement("INSERT INTO source VALUES (1, 'Mike', 'HR'), (2, 'Linda', 'Finance')");
```

Comment:
Why not use `testTables.createTable` methods?

Reply:
It's because I wanted to use a regular Hive table as the source here (since that should be a typical CTAS use case for Iceberg)

Reply:
Got it. The we might want to create a method for this as well, since we have this code at least in 3 places now

Reply:
Sure, makes sense